app/src/main/java/org/thoughtcrime/securesms/conversation/start/NewConversationHomeFragment.kt

```diff
@@ -88,30 +94,47 @@ fun NewConversationScreen(
 ) {
     Column(modifier = Modifier.background(LocalColors.current.backgroundSecondary)) {
         AppBar(stringResource(R.string.dialog_new_conversation_title), onClose = delegate::onDialogClosePressed)
-        ItemButton(textId = R.string.messageNew, icon = R.drawable.ic_message, onClick = delegate::onNewMessageSelected)
-        Divider(startIndent = LocalDimensions.current.dividerIndent)
-        ItemButton(textId = R.string.activity_create_group_title, icon = R.drawable.ic_group, onClick = delegate::onCreateGroupSelected)
-        Divider(startIndent = LocalDimensions.current.dividerIndent)
-        ItemButton(textId = R.string.dialog_join_community_title, icon = R.drawable.ic_globe, onClick = delegate::onJoinCommunitySelected)
-        Divider(startIndent = LocalDimensions.current.dividerIndent)
-        ItemButton(textId = R.string.activity_settings_invite_button_title, icon = R.drawable.ic_invite_friend, contentDescription = R.string.AccessibilityId_invite_friend_button, onClick = delegate::onInviteFriend)
-        Column(
-            modifier = Modifier
-                .padding(horizontal = LocalDimensions.current.marginMedium)
-                .padding(top = LocalDimensions.current.itemSpacingMedium)
+        Surface(
+            modifier = Modifier.nestedScroll(rememberNestedScrollInteropConnection())
         ) {
-            Text(
-                text = stringResource(R.string.accountIdYours),
-                style = xl
-            )
-            Spacer(modifier = Modifier.height(LocalDimensions.current.itemSpacingTiny))
-            Text(
-                text = stringResource(R.string.qrYoursDescription),
-                color = LocalColors.current.textSecondary,
-                style = small
-            )
-            Spacer(modifier = Modifier.height(LocalDimensions.current.itemSpacingSmall))
-            QrImage(string = accountId, Modifier.contentDescription(R.string.AccessibilityId_qr_code))
+            Column(
+                modifier = Modifier.verticalScroll(rememberScrollState())
+            ) {
+                Items(accountId, delegate)
+            }
         }
     }
 }
+
+/**
+ * Items of the NewConversationHome screen. Use in a [Column]
+ */
+@Suppress("UnusedReceiverParameter")
+@Composable
+fun ColumnScope.Items(
```

Comment:
This adds a new method to ALL columns _throughout_ the app... Items in this case are deeply tied to this screen.

Reply:
> This adds a new method to ALL columns throughout the app.

not really, it just limits this function to being allowed to be called from inside a `Column`. If it was not an extension function you could call it from anywhere, so yeah, I can probably make a lot of these functions private or internal.

> Items in this case are deeply tied to this screen.

Yeah, if you don't call `Items()` in `Column` or similar, then they will not layout properly. But if it was LazyList then this function would look much different, and I don't know another `@Composable` than `Column` that I'd want to allow this to be used in.

Reply:
I'm not saying it shouldn't be a Column, I'm saying that you have created something that can now be accessed by _all_ columns anywhere in the app.
If I'm in another screen, I'll be able to cal `Items` which won't mean anything in that screen, as the thing you called Items are items of the `NewConversationHome`. And yet they will exist everywhere (in all Columns).
I could call this in `LoadAccountScreen` for example.
If you are creating a composable that needs thigns laid out as a column then you should simply add a column at the root of your `Items(` composable, and remove the extension from ColumnScope so that it can't be accessed anywhere inside a column in the app

Reply:
@bemusementpark

Reply:
fixed. I made it `private`. It's now only available in a `Column` **in this file.**